File: kotlin/src/main/kotlin/advertisements/user/domain/UserRepository.kt
package advertisements.user.domain

import advertisements.shared.value_object.UserId

interface UserRepository {
    fun findAdminById(id: UserId): AdminUser?
    fun findMemberById(id: UserId): MemberUser?
    fun findAdminOrMemberById(id: UserId): Any? // AdminUser or MemberUser or null
    fun saveMember(member: MemberUser)
}

File: typescript/src/advertisements/user/domain/UserRepository.ts
import { AdminUser } from './AdminUser';
import { MemberUser } from './MemberUser';
import {UserId} from "../../shared/domain/value-object/UserId";

export interface UserRepository {
    // findAdminById(id: UserId): Promise<AdminUser | null>;
    findMemberById(id: UserId): Promise<MemberUser | null>;
    // findAdminOrMemberById(id: UserId): Promise<AdminUser | MemberUser | null>;
    // saveMember(member: MemberUser): Promise<void>;
}
